tests/test_buku.py

```diff
@@ -26,20 +26,58 @@ def check_import_html_results_contains(result, expected_result):
     return count == n * (n + 1) / 2
 
 
-@pytest.mark.parametrize(
-    "url, exp_res",
-    [
-        ["http://example.com", False],
-        ["ftp://ftp.somedomain.org", False],
-        ["http://examplecom.", True],
-        ["http://.example.com", True],
-        ["http://example.com.", True],
-        ["about:newtab", True],
-        ["chrome://version/", True],
-    ],
-)
+@pytest.mark.parametrize('url, result', [
+    ('http://user:password@hostname:1234/path?query#hash', 'user:password'),
+    ('http://:password@hostname:1234/path?query#hash', ':password'),
+    ('http://user:@hostname:1234/path?query#hash', 'user:'),
+    ('http://user@hostname:1234/path?query#hash', 'user'),
+    ('http://@hostname:1234/path?query#hash', ''),
+    ('http://hostname:1234/path?query#hash', None),
+    ('//[', ValueError('Invalid IPv6 URL')),
+    ('//⁈', ValueError("netloc '⁈' contains invalid characters under NFKC normalization")),
+])
+def test_extract_auth(url, result):
+    if not isinstance(result, Exception):
+        assert extract_auth(url) == (result, 'http://hostname:1234/path?query#hash')
+    else:
+        try:
+            extract_auth(url)
+        except Exception as e:
+            assert repr(e) == repr(result)
+        else:
+            assert False, f'expected {repr(result)} to be raised'
```

Comment:
Thoroughly testing URL parsing here (along with auth extraction).

Also included [examples of `urlparse()` errors](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse):
![image](https://github.com/user-attachments/assets/beb70e50-85fb-4d47-a928-41035b0f0ebf)